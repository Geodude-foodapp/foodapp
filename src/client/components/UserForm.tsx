import React, { useState } from 'react';
import { UserFormState } from '../../Types';
import { useNavigate } from 'react-router-dom';

type UserFormProps = {
  type: 'Sign Up' | 'Log In';
  handleSubmit: () => void;
};

export default ({ type, handleSubmit }: UserFormProps) => {
  const initialFormState: UserFormState = {
    name: '',
    password: '',
  };

  const navigate = useNavigate();
  const [formData, setFormData] = useState(initialFormState);

  return (
    <section id='user-form'>
      <h2>{`${type} Form`}</h2>
      <form onSubmit={handleSubmit}>
        <input
          type='text'
          placeholder='username'
          value={formData.name}
          onChange={(e) =>
            setFormData((state) => ({ ...state, username: e.target.value }))
          }
        />
        <input
          type='password'
          placeholder='password'
          value={formData.password}
          onChange={(e) =>
            setFormData((state) => ({ ...state, password: e.target.value }))
          }
        />
        {type === 'Sign Up' && (
          <>
            <legend>Select your diet:</legend>
            <div>
              <input
                type='checkbox'
                name='vegan'
              ></input>
            </div>
          </>
        )}
        <button>Submit</button>
      </form>
    </section>
  );
};
