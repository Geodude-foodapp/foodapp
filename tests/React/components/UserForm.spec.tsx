import { render } from '@testing-library/react';
import React from 'react';
import UserForm from '../../../src/client/components/UserForm';

const handleSubmit = jest.fn();

describe('Sign Up Form Component', () => {
  beforeAll(() =>
    render(
      <UserForm
        type='Sign Up'
        handleSubmit={handleSubmit}
      />
    )
  );
  it('matches the snapshot', () => {
    const userForm = document.querySelector('#user-form');
    expect(userForm).toMatchSnapshot();
  });
});

describe('Log In Form Component', () => {
  beforeAll(() =>
    render(
      <UserForm
        type='Log In'
        handleSubmit={handleSubmit}
      />
    )
  );
  it('matches the snapshot', () => {
    const userForm = document.querySelector('#user-form');
    expect(userForm).toMatchSnapshot();
  });
});
