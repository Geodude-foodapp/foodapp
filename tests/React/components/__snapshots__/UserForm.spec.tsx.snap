// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Log In Form Component matches the snapshot 1`] = `
<section
  id="user-form"
>
  <h2>
    Log In Form
  </h2>
  <form>
    <input
      placeholder="username"
      type="text"
      value=""
    />
    <input
      placeholder="password"
      type="password"
      value=""
    />
    <button>
      Submit
    </button>
  </form>
</section>
`;

exports[`Sign Up Form Component matches the snapshot 1`] = `
<section
  id="user-form"
>
  <h2>
    Sign Up Form
  </h2>
  <form>
    <input
      placeholder="username"
      type="text"
      value=""
    />
    <input
      placeholder="password"
      type="password"
      value=""
    />
    <button>
      Submit
    </button>
  </form>
</section>
`;
